projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/services/data-jobs-base.api.service.ts

```diff
@@ -0,0 +1,117 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { Injectable } from '@angular/core';
+
+import { Observable } from 'rxjs';
+
+import { ApolloQueryResult, DefaultOptions, gql, InMemoryCache } from '@apollo/client/core';
+
+import { Apollo, ApolloBase, QueryRef } from 'apollo-angular';
+import { HttpLink } from 'apollo-angular/http';
+
+import { DataJob, DataJobExecutionsPage, DataJobExecutionsReqVariables, DataJobPage, DataJobReqVariables } from '../model';
+
+/**
+ * ** Data Jobs Service build on top of Apollo gql client.
+ */
+@Injectable()
+export class DataJobsBaseApiService {
+    private static readonly APOLLO_METHOD = 'GET';
+    private static readonly APOLLO_DEFAULT_OPTIONS: DefaultOptions = {
+        watchQuery: {
+            fetchPolicy: 'no-cache',
+            errorPolicy: 'all'
+        },
+        query: {
+            fetchPolicy: 'no-cache',
+            errorPolicy: 'all'
+        }
+    };
+
+    /**
+     * ** Constructor.
+     */
+    constructor(private readonly apollo: Apollo,
+                private readonly httpLink: HttpLink) {
+    }
+
+    /**
+     * ** Get all DataJobs for provided OwnerTeam and load data based on provided gqlQuery.
+     */
+    getJobs(
+        ownerTeam: string,
+        gqlQuery: string,
+        variables: DataJobReqVariables): Observable<ApolloQueryResult<DataJobPage>> {
+
+        return this.getApolloClientFor(ownerTeam)
+                   .query({
+                       query: gql`${ gqlQuery }`,
+                       variables
+                   });
+    }
+
+    /**
+     * ** Create Apollo watcher for gqlQuery.
+     */
+    watchForJobs(
+        ownerTeam: string,
+        gqlQuery: string,
+        variables: DataJobReqVariables): QueryRef<DataJobPage, DataJobReqVariables> {
+
+        return this.getApolloClientFor(ownerTeam)
+                   .watchQuery({
+                       query: gql`${ gqlQuery }`,
+                       variables
+                   });
+    }
+
+    /**
+     * ** Get all DataJob Executions for provided OwnerTeam and load data based on provided gqlQuery.
+     */
+    getExecutions( // NOSONAR
+        ownerTeam: string,
+        gqlQuery: string,
+        variables: DataJobExecutionsReqVariables
+    ): Observable<ApolloQueryResult<DataJobExecutionsPage>> {
+
+        return this.getApolloClientFor(ownerTeam)
+                   .query({
+                       query: gql`${ gqlQuery }`,
+                       variables
+                   });
+    }
+
+    private getApolloClientFor(ownerTeam: string): ApolloBase<DataJob> {
+        if (!this.apollo.use(ownerTeam)) {
+            this.apollo.createNamed(ownerTeam, {
+                cache: new InMemoryCache({
+                    typePolicies: {
+                        Query: {
+                            fields: {
+                                // TODO workaround documented in https://jira.eng.vmware.com/browse/TAUR-4179
```

Comment:
Please, remove this internal reference: jira.eng.vmware.com. Also search around to see if we have this somewhere else in the code.